File: src/main/java/cloudator/WeatherServiceUpdater.java
package cloudator;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.env.Environment;
import org.springframework.scheduling.TaskScheduler;
import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;

@Component
public class WeatherServiceUpdater {

    @Autowired
    TaskScheduler taskScheduler;


    @Autowired
    WeatherServiceExecutor task;

    @Autowired
    Environment env;

    @PostConstruct
    public void init() {
        int schedulingInterval = env.getProperty("interval.sec", Integer.class, 60);
        System.out.println("Scheduling  every " + schedulingInterval);
        taskScheduler.scheduleWithFixedDelay(task, schedulingInterval * 1000);
    }
}

File: src/main/scala/cloudator/WeatherServiceUpdater.scala
package cloudator

import javax.annotation.PostConstruct

import cloudator.util.Logging
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.env.Environment
import org.springframework.scheduling.TaskScheduler
import org.springframework.stereotype.Component

import scala.util.Try

@Component
class WeatherServiceUpdater extends Logging {
  @Autowired private[cloudator] val taskScheduler: TaskScheduler = null
  @Autowired private[cloudator] val task: WeatherServiceExecutor = null
  @Autowired private[cloudator] val env: Environment = null

  @PostConstruct
  def init(): Unit = {
    val schedulingInterval = Try(env.getProperty("interval.sec").toInt).getOrElse(120)
    logInfo("Scheduling  every " + schedulingInterval + " sec")
    taskScheduler.scheduleWithFixedDelay(task, schedulingInterval * 1000)
  }
}